test/integration/puppeteer.js

```diff
@@ -0,0 +1,156 @@
+
+    // prepare page
+    await page.evaluate(async (file, pageSize, minPageSize, maxPageSize, networkTax) => {
+      let button = document.getElementById('startButton');
+      if (button) button.click();
+      let style = document.createElement('style');
+      style.type = 'text/css';
+      style.innerHTML = `body { font size: 0 !important; }
+                         #info, button, input, body > div.dg.ac, body > div.lbl { display: none !important; }`;
+      document.getElementsByTagName('head')[0].appendChild(style);
+      let canvas = document.getElementsByTagName('canvas');
+      for (let i = 0; i < canvas.length; i++) {
+        if (canvas[i].height == 48) canvas[i].style.display = 'none';
+      }
+      if (file == 'misc_animation_authoring') {
+        let divs = document.getElementsByTagName('div')
+        for(let i = 0; i < divs.length; i++) divs[i].style.display ='none';
+      }
+      let resourcesSize = (pageSize / 1024 / 1024 - minPageSize) / maxPageSize;
+      await new Promise((resolve, _) => setTimeout(resolve, networkTax * resourcesSize));
+    }, file, pageSize, minPageSize, maxPageSize, networkTax);
+
+    // render promise
+    await page.evaluate(async (renderTimeout, renderInterval) => {
+      window.renderStarted = true;
+      await new Promise(function(resolve, _) {
+        let renderStart = performance.wow();
+        let waitingLoop = setInterval(function() {
+          let renderEcceded = (performance.wow() - renderStart > renderTimeout * window.maxFrameId);
+          if (window.renderFinished || renderEcceded) {
+            if (renderEcceded) console.log('Render timeout exceeded...');
+            clearInterval(waitingLoop);
+            resolve();
+          }
+        }, renderInterval);
+      });
+    }, renderTimeout, renderInterval);
+
+    if (process.env.GENERATE) {
+
+      // generate screenshots
+      await page.screenshot({ path: `./examples/screenshots/${file}.png` });
+      console.greenLog(`file: ${file} generated`);
+
+    } else if (fs.existsSync(`./examples/screenshots/${file}.png`)) {
+
+      // diff screenshots
+      await page.screenshot({ path: `./node_modules/temp.png` });
+      let img1 = PNG.sync.read(fs.readFileSync(`./examples/screenshots/${file}.png`));
+      let img2 = PNG.sync.read(fs.readFileSync(`./node_modules/temp.png`));
+      let diff = new PNG({ width: img1.width, height: img1.height });
+      try {
+        pixelmatch(img1.data, img2.data, diff.data, img1.width, img1.height, { threshold: threshold });
+      } catch(e) {
+        ++failedScreenshot;
+        console.redLog(`ERROR! Image sizes does not match in file: ${file}`)
+      }
+
+      // save and print result
+      let stream = fs.createWriteStream(`./node_modules/diff.png`);
+      diff.pack().pipe(stream);
+      stream.end();
+      let currDiff = diff.data
+        .filter((bit, i) => (i % 4 == 0) && (bit == 255) && (diff.data[i+1] == 0) && (diff.data[i+2] == 0))
```

Comment:
It would be great if you can format both new JS files according to the project's style guide. Also not that the project prefers `===` over `==`.

Reply:
Yep, but right now POC work only in my repo and local machine. Probably travis can't run expess right now. We already have http-server for this purpose, but the problem that `concurrently` gives a lot of additional noize in console.

Reply:
Um, how is this related to the code style? I was referring to https://zz85.github.io/mrdoobapproves/

Reply:
Yep, I will fix code style later.